sql/core/src/test/scala/org/apache/spark/sql/CachedTableSuite.scala

```diff
@@ -522,146 +522,152 @@ class CachedTableSuite extends QueryTest with SQLTestUtils
   }
 
   test("A cached table preserves the partitioning and ordering of its cached SparkPlan") {
-    val table3x = testData.union(testData).union(testData)
-    table3x.createOrReplaceTempView("testData3x")
-
-    sql("SELECT key, value FROM testData3x ORDER BY key").createOrReplaceTempView("orderedTable")
-    spark.catalog.cacheTable("orderedTable")
-    assertCached(spark.table("orderedTable"))
-    // Should not have an exchange as the query is already sorted on the group by key.
-    verifyNumExchanges(sql("SELECT key, count(*) FROM orderedTable GROUP BY key"), 0)
-    checkAnswer(
-      sql("SELECT key, count(*) FROM orderedTable GROUP BY key ORDER BY key"),
-      sql("SELECT key, count(*) FROM testData3x GROUP BY key ORDER BY key").collect())
-    uncacheTable("orderedTable")
-    spark.catalog.dropTempView("orderedTable")
-
-    // Set up two tables distributed in the same way. Try this with the data distributed into
-    // different number of partitions.
-    for (numPartitions <- 1 until 10 by 4) {
+    // Set MAX_SINGLE_PARTITION_BYTES to Long.MaxValue to avoid inserting Exchange node
```

Comment:
If we need to change this, this looks like a side-effect of this change. What is the changed `assert` statement in the test code, @pan3793 ? For me, `numPartitions` seems to be still 6 and 12 in the same line (before and after).

Reply:
`MAX_SINGLE_PARTITION_BYTES=128m` causes the plan change seems reasonable to me, WDYT ?@wangyum 

And the only change here is 

```patch
+ withSQLConf(SQLConf.MAX_SINGLE_PARTITION_BYTES.key -> Long.MaxValue.toString) {
    // original code with additional two leading spaces
+ }
```

Reply:
Ah, got it. I was confused. BTW, the test name is `test("A cached table preserves the partitioning and ordering of its cached SparkPlan"`. So, after this PR, Spark doesn't preserve the partitioning and ordering of its cached SparkPlan?

Reply:
No, still valid. With `MAX_SINGLE_PARTITION_BYTES=128m`, the assertion in `verifyNumExchanges` may fail because it may insert new Exchange nodes.